src/content/index.ts

```diff
@@ -0,0 +1,47 @@
+import browser from 'webextension-polyfill'
+import type { MessageType } from '../types'
+
+const KEY = 'LOCALE_SWITCHER_LANGUAGE'
+
+function embedScript() {
+  const locale = window.localStorage.getItem(KEY)
+  if (!locale) return
+
+  const code = `
+    (() => {
+      Object.defineProperties(window.navigator.__proto__, {
+        language: {
+          value: '${locale}',
+          enumerable: true,
+        },
+        languages: {
+          value: ['${locale}'],
+          enumerable: true,
+        }
+      });
+    })();`
+
+  const script = document.createElement('script')
+  script.textContent = code
+  document.documentElement.prepend(script)
+  script.remove()
+}
+
+function handleMessage({ type, data }: MessageType) {
+  switch (type) {
+    case 'update':
+      if (data === window.localStorage.getItem(KEY)) break
```

Comment:
In my testing in Chrome `window.localStorage` was not accessible inside extension event listeners. This is why I had to use the same injection trick above to change the localStorage / sessionStorage. Did this work for you in Chrome?

https://github.com/locale-switcher/locale-switcher/commit/d7ace06e091e91d179cb4e5b9c770e9d42d6820e#diff-c89f55ed43133253050dbce66bfd2c8a4f6280615dbaa582b7efbe3104a79614R17

Reply:
Mh.. it works for me in chrome?